[
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "<!--e57a25ab6845829454e8d69fc972939a-->\n\nThe following sections might be updated with supplementary metadata relevant to reviewers and maintainers.\n\n<!--006a51241073e994b41acfe9ec718e94-->\n### Code Coverage\nFor detailed information about the code coverage, see the [test coverage report](https://corecheck.dev/bitcoin/bitcoin/pulls/30105).\n<!--021abf342d371248e50ceaed478a90ca-->\n### Reviews\nSee [the guideline](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#code-review) for information on the review process.\nA summary of reviews will appear here.\n",
      "created_at" : "2024-05-15T00:03:17Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/30105#issuecomment-2111350303",
      "id" : 2111350303,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/30105",
      "node_id" : "IC_kwDOABII58592KYf",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/2111350303/reactions"
      },
      "updated_at" : "2024-05-15T00:03:18Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/2111350303",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/39886733?v=4",
         "events_url" : "https://api.github.com/users/DrahtBot/events{/privacy}",
         "followers_url" : "https://api.github.com/users/DrahtBot/followers",
         "following_url" : "https://api.github.com/users/DrahtBot/following{/other_user}",
         "gists_url" : "https://api.github.com/users/DrahtBot/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/DrahtBot",
         "id" : 39886733,
         "login" : "DrahtBot",
         "node_id" : "MDQ6VXNlcjM5ODg2NzMz",
         "organizations_url" : "https://api.github.com/users/DrahtBot/orgs",
         "received_events_url" : "https://api.github.com/users/DrahtBot/received_events",
         "repos_url" : "https://api.github.com/users/DrahtBot/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/DrahtBot/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/DrahtBot/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/DrahtBot"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "<!--85328a0da195eb286784d51f73fa0af9-->\n\nð§ At least one of the CI tasks failed. Make sure to run all tests locally, according to the\ndocumentation.\n\nPossibly this is due to a silent merge conflict (the changes in this pull request being\nincompatible with the current code in the target branch). If so, make sure to rebase on the latest\ncommit of the target branch.\n\nLeave a comment here, if you need help tracking down a confusing failure.\n\n<sub>Debug: https://github.com/bitcoin/bitcoin/runs/24977593545</sub>",
      "created_at" : "2024-05-15T00:59:42Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/30105#issuecomment-2111396714",
      "id" : 2111396714,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/30105",
      "node_id" : "IC_kwDOABII58592Vtq",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/2111396714/reactions"
      },
      "updated_at" : "2024-05-15T00:59:42Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/2111396714",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/39886733?v=4",
         "events_url" : "https://api.github.com/users/DrahtBot/events{/privacy}",
         "followers_url" : "https://api.github.com/users/DrahtBot/followers",
         "following_url" : "https://api.github.com/users/DrahtBot/following{/other_user}",
         "gists_url" : "https://api.github.com/users/DrahtBot/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/DrahtBot",
         "id" : 39886733,
         "login" : "DrahtBot",
         "node_id" : "MDQ6VXNlcjM5ODg2NzMz",
         "organizations_url" : "https://api.github.com/users/DrahtBot/orgs",
         "received_events_url" : "https://api.github.com/users/DrahtBot/received_events",
         "repos_url" : "https://api.github.com/users/DrahtBot/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/DrahtBot/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/DrahtBot/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/DrahtBot"
      }
   },
   {
      "_links" : {
         "html" : {
            "href" : "https://github.com/bitcoin/bitcoin/pull/30105#discussion_r1600847551"
         },
         "pull_request" : {
            "href" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/30105"
         },
         "self" : {
            "href" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/1600847551"
         }
      },
      "author_association" : "CONTRIBUTOR",
      "body" : "Made a temporary modification of a word in the expected message (of `feature_assumeutxo.py` to test that a mismatch would be caught (it was, as expected).  Restored to original expected message then modified the message in `validation.cpp`.  This mismatch was also detected (as expected)l.\r\n\r\nhttps://github.com/bitcoin/bitcoin/blob/695d80126f75a99c80d5d6c18ebaaabf7410b989/src/validation.cpp#L5659\r\n\r\nThe default timeout for `assert_debug_log()` is 2 seconds.  I haven't had a chance to dig deeper yet, but are we confident there isn't a potential timing issue with the message arriving into the debug log later than 2 seconds?  (e.g. a late arrival potentially causing an intermittent test failure).\r\n\r\nhttps://github.com/bitcoin/bitcoin/blob/695d80126f75a99c80d5d6c18ebaaabf7410b989/test/functional/test_framework/test_node.py#L462",
      "commit_id" : "de7ac866cc2d1998e6fc171c23cc9c6ef8b66016",
      "created_at" : "2024-05-15T01:56:02Z",
      "diff_hunk" : "@@ -237,6 +237,9 @@ def run_test(self):\n         # will allow us to test n1's sync-to-tip on top of a snapshot.\n         self.generate(n0, nblocks=100, sync_fun=self.no_op)\n \n+        with n0.assert_debug_log(expected_msgs=[\"[snapshot] activation failed - work does not exceed active chainstate\",\"[snapshot] activation failed - population failed\"]):",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/30105#discussion_r1600847551",
      "id" : 1600847551,
      "line" : 240,
      "node_id" : "PRRC_kwDOABII585fav6_",
      "original_commit_id" : "de7ac866cc2d1998e6fc171c23cc9c6ef8b66016",
      "original_line" : 240,
      "original_position" : 4,
      "original_start_line" : null,
      "path" : "test/functional/feature_assumeutxo.py",
      "position" : 4,
      "pull_request_review_id" : 2056736096,
      "pull_request_url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/30105",
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/1600847551/reactions"
      },
      "side" : "RIGHT",
      "start_line" : null,
      "start_side" : null,
      "subject_type" : "line",
      "updated_at" : "2024-05-15T02:06:19Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/1600847551",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/106488469?v=4",
         "events_url" : "https://api.github.com/users/tdb3/events{/privacy}",
         "followers_url" : "https://api.github.com/users/tdb3/followers",
         "following_url" : "https://api.github.com/users/tdb3/following{/other_user}",
         "gists_url" : "https://api.github.com/users/tdb3/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/tdb3",
         "id" : 106488469,
         "login" : "tdb3",
         "node_id" : "U_kgDOBljilQ",
         "organizations_url" : "https://api.github.com/users/tdb3/orgs",
         "received_events_url" : "https://api.github.com/users/tdb3/received_events",
         "repos_url" : "https://api.github.com/users/tdb3/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/tdb3/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/tdb3/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/tdb3"
      }
   },
   {
      "_links" : {
         "html" : {
            "href" : "https://github.com/bitcoin/bitcoin/pull/30105#discussion_r1600849721"
         },
         "pull_request" : {
            "href" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/30105"
         },
         "self" : {
            "href" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/1600849721"
         }
      },
      "author_association" : "CONTRIBUTOR",
      "body" : "Also made a temporary modification to this expected message.  Mismatch was caught (as expected).",
      "commit_id" : "de7ac866cc2d1998e6fc171c23cc9c6ef8b66016",
      "created_at" : "2024-05-15T02:00:54Z",
      "diff_hunk" : "@@ -237,6 +237,9 @@ def run_test(self):\n         # will allow us to test n1's sync-to-tip on top of a snapshot.\n         self.generate(n0, nblocks=100, sync_fun=self.no_op)\n \n+        with n0.assert_debug_log(expected_msgs=[\"[snapshot] activation failed - work does not exceed active chainstate\",\"[snapshot] activation failed - population failed\"]):\n+            assert_raises_rpc_error(-32603, \"Unable to load UTXO snapshot\", n0.loadtxoutset, dump_output['path'])",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/30105#discussion_r1600849721",
      "id" : 1600849721,
      "line" : 241,
      "node_id" : "PRRC_kwDOABII585fawc5",
      "original_commit_id" : "de7ac866cc2d1998e6fc171c23cc9c6ef8b66016",
      "original_line" : 241,
      "original_position" : 5,
      "original_start_line" : null,
      "path" : "test/functional/feature_assumeutxo.py",
      "position" : 5,
      "pull_request_review_id" : 2056736096,
      "pull_request_url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/30105",
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/1600849721/reactions"
      },
      "side" : "RIGHT",
      "start_line" : null,
      "start_side" : null,
      "subject_type" : "line",
      "updated_at" : "2024-05-15T02:04:55Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/1600849721",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/106488469?v=4",
         "events_url" : "https://api.github.com/users/tdb3/events{/privacy}",
         "followers_url" : "https://api.github.com/users/tdb3/followers",
         "following_url" : "https://api.github.com/users/tdb3/following{/other_user}",
         "gists_url" : "https://api.github.com/users/tdb3/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/tdb3",
         "id" : 106488469,
         "login" : "tdb3",
         "node_id" : "U_kgDOBljilQ",
         "organizations_url" : "https://api.github.com/users/tdb3/orgs",
         "received_events_url" : "https://api.github.com/users/tdb3/received_events",
         "repos_url" : "https://api.github.com/users/tdb3/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/tdb3/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/tdb3/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/tdb3"
      }
   }
]
